front/lib/auth.ts

```diff
@@ -383,10 +407,12 @@ export class Authenticator {
       throw new Error(`Could not find workspace with sId ${workspaceId}`);
     }
 
+    let globalGroup: GroupResource | null = null;
     let subscription: SubscriptionType | null = null;
     let flags: WhitelistableFeature[] = [];
 
-    [subscription, flags] = await Promise.all([
+    [globalGroup, subscription, flags] = await Promise.all([
+      GroupResource.internalFetchWorkspaceGlobalGroup(workspace.id),
```

Comment:
I think internalAdminForWorkspace really assume access to everything no? Should really be internalSystemForWorkspace which gives a system API. Let's file some tasks to not forget to figure that all out.

Reply:
I added a comment and I'll add a task. 👍🏻